src/magnify/registry.py

```diff
@@ -29,11 +32,254 @@ def component_factory(*args, **kwargs):
+    debug :
+        If True, set the logger to debug mode for error logging.
+    shape : tuple[int, int], optional
+        The shape of the button array, specifying the number of rows and columns in the image grid.
+        If provided, it will be used to assign default button tags. The default is `(8, 8)` for mini chips.
+    pinlist :
+        A file path to a CSV file that describes the pin layout on the chip. The CSV file must include
+        a column called `Indices` that contains row and column pairs in the format `(row, col)`, and a
+        `MutantID` column that contains the names of the buttons. This is used to map the buttons to physical
+        locations on the chip. Either `pinlist` or `shape` must be provided.
+    blank : 
+        Values representing "blank" or non-expressed buttons in the pinlist, which will be replaced
+        with empty strings in the dataset. Defaults to `["", "blank", "BLANK"]`.
+    overlap :
+        The number of pixels to exclude from the edges of adjacent tiles during the stitching process.
+        This overlap value is subtracted from both the vertical (y) and horizontal (x) dimensions
+        of the tiles to remove redundant or overlapping areas between adjacent tiles.
+    rotation : 
+        The degree of rotation to apply to the stitched image.
+    row_dist / col_dist: 
+        The distance between rows/columns of buttons in pixels. This is converted to pixels based on the pixel-to-micron conversion rate, assuming 1.61 pixels.
+    min_button_radius / max_button_radius : 
+        The minimum/maximum radius (in pixels) for detecting buttons in the image.
+    low_edge_quantile / high_edge_quantile : 
+        The lower/upper quantile for edge detection, used to identify the dimmest edges when detecting buttons.
+    num_iter : 
+        The maximum number of iterations to perform during the button detection process, controlling the depth of search.
+    min_roundness : 
+        The minimum roundness value for detected buttons. Buttons that do not meet this roundness threshold are excluded.
+    cluster_penalty : 
+        The penalty factor for clustering during button detection. This parameter helps control how strongly nearby buttons are grouped together.
+    roi_length : 
+        The length (in pixels) of the region of interest (ROI) around detected buttons.
+    progress_bar : 
+        If True, display a progress bar during processing to track the progress of the pipeline.
+    search_timestep : 
+        A list of time steps to focus on for button detection. If provided, only the specified time steps are used. If `None`, all time steps are used.
+    search_channel : 
+        The channel or list of channels to use for button detection and expression filtering. If `None`, all channels will be used. Channels are used to compute foreground and background values for contrast-based filtering.
```

Comment:
Remove "Channels are used to compute foreground and background values for contrast-based filtering." you already provide enough info in the earlier part of the docstring.

Reply:
Resolved